### PR TITLE
New RC modules

### PR DESCRIPTION
**Pull request by KhrystynaDubovyk**
**Enums** index - add new params to PrimaryAudioSource, ModuleType enums; add new enums LightName, LightStatus, DisplayMode, DistanceUnit.
**Structs** index - add/update ClimateControlCapabilities, ClimateControlData, RadioControlCapabilities, StationIDNumber, SisData, RadioControlData, AudioControlCapabilities, EqualizerSettings, AudioControlData, LightCapabilities, LightControlCapabilities, LightState, LightControlData, HMISettingsControlCapabilities, HMISettingsControlData, RemoteControlCapabilities.
**GetInteriorVehicleData** – update Request, Response sections
**OnInteriorVehicleData** – update description.

Technical task :  #[1798](https://github.com/smartdevicelink/sdl_core/issues/1798)
This PR is **ready** for review.

This pull request contains the implementation of new remote control modules and parameters
(proposal: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0099-new-remote-control-modules-and-parameters.md)
For testing use the following test set:
https://github.com/smartdevicelink/sdl_atf_test_scripts/blob/feature/rc_new_modules/test_sets/rc_AUDIO_LIGHT_HMI_SETTINGS.txt

Pull request with ATF scripts: https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1909

For manual testing use:
HMI from : smartdevicelink/sdl_hmi#52
mobile application SPT from HockeyApp 20180213-Nightly-Android

### Risk
This PR makes **[major]** API changes.
### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: docs/Common/Structs/index.md [hunk 1]
```diff
@@ -540,6 +584,92 @@
...
+|channelName|String|false|maxlength="50"|Read-only channel / frequency name (e.i. "Treble, Midrange, Bass" or "125 Hz")|
+|channelSetting|Integer|true|minvalue="0" <br> maxvalue="100"|Reflects the setting, from 0%-100%.|
+
+### AudioControlData
+|Name|Type|Mandatory|Additional|Description|
+|:---|:---|:--------|:---------|:----------|
+|source|Common.PrimaryAudioSource|false||In a getter response or a notification, it is the current primary audio source of the system. <br> In a setter request, it is the target audio source that the system shall switch to. <br> If the value is MOBILE_APP, the system shall switch to the mobile media app that issues the setter RPC.|
+|keepContext|Boolean|false||This parameter shall not be present in any getter responses or notifications. <br> This parameter is optional in a setter request. The default value is false. <br> If it is true, the system not only changes the audio source but also brings the default infotainment system UI associated with the audio source to foreground and set the application to background. <br> If it is false, the system changes the audio source, but keeps the current application's context.|
```
Comment (yang1070) [inline docs/Common/Structs/index.md]: <br> If it is `false`, the system not only changes the audio source but also brings the default infotainment system UI associated with the audio source to foreground and set the application to background. <br> If it is `true`, the system changes the audio source, but keeps the current application's context.|

Comment (KhrystynaDubovyk) [reply docs/Common/Structs/index.md]: @yang1070, thank you.  
The description of the `keepContext` param  
provided in the proposal [0099-new-remote-control-modules-and-parameters.md](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0099-new-remote-control-modules-and-parameters.md)  
section **New AUDIO data types** says:  
```xml  
    <param name="keepContext" type="Boolean" mandatory="false">
      <description>
      This parameter shall not be present in any getter responses or notifications.
      This parameter is optional in a setter request. The default value is false.
      If it is true, the system not only changes the audio source but also brings the default infotainment system UI associated with the audio source to foreground and set the application to background.
      If it is false, the system changes the audio source, but keeps the current application's context.
      </description>  
```
@yang1070, please confirm updating the description according to your comment above.

**Comment by yang1070** [comment]
@theresalech  Ford has reviewed and approved this PR, It is ready for Livio to review.

**Comment by AStasiuk** [comment]
@yang1070 could you please review the latest changes?

File: docs/Common/Enums/index.md [hunk 2]
```diff
@@ -925,6 +941,54 @@ Radio Buttons
...
+|FRONT_LEFT_DAYTIME_RUNNING_LIGHT|8||
+|FRONT_RIGHT_DAYTIME_RUNNING_LIGHT|9||
+|FRONT_LEFT_TURN_LIGHT|10||
+|FRONT_RIGHT_TURN_LIGHT|11||
+|REAR_LEFT_FOG_LIGHT|12||
+|REAR_RIGHT_FOG_LIGHT|13||
+|REAR_LEFT_TAIL_LIGHT|14||
+|REAR_RIGHT_TAIL_LIGHT|15||
+|REAR_LEFT_BREAK_LIGHT|16||
+|REAR_RIGHT_BREAK_LIGHT|17||
```
Comment (jacobkeeler) [inline docs/Common/Enums/index.md]: `BRAKE_LIGHT`

File: docs/Common/Enums/index.md [hunk 3]
```diff
@@ -230,6 +230,17 @@
...
+### DeactivateReason
+
+|Name|Value|Description|
+|:---|:----|:----------|
+|AUDIO|0||
+|PHONECALL|1||
+|NAVIGATIONMAP|2||
+|PHONEMENU|3||
+|SYNCSETTINGS|4||
+|GENERAL|5||
```
Comment (jacobkeeler) [inline docs/Common/Enums/index.md]: Doesn't seem relevant to the feature.

File: docs/Common/Structs/index.md [hunk 4]
```diff
@@ -558,15 +673,18 @@
 |stateAvailable|Boolean|false|| Availability of the getting the Radio state. <br> True: Available, False: Not Available, Not present: Not Available.|
 |signalStrengthAvailable|Boolean|false||Availability of the getting the signal strength. <br> True: Available, False: Not Available, Not present: Not Available.|
 |signalChangeThresholdAvailable|Boolean|false||Availability of the getting the signal Change Threshold. <br> True: Available, False: Not Available, Not present: Not Available.|
+|sisDataAvailable|Boolean|false|| Availability of the getting HD radio Station Information Service (SIS) data. <br> True: Available, False: Not Available, Not present: Not Available.|
```
Comment (jacobkeeler) [inline docs/Common/Structs/index.md]: `sisData` is missing from `RadioControlData`

File: docs/Common/Enums/index.md [hunk 5]
```diff
@@ -664,6 +675,8 @@ Radio Buttons
 |LINE_IN|4||
 |IPOD|5||
 |MOBILE_APP|6||
+|CD|7||
```
Comment (jacobkeeler) [inline docs/Common/Enums/index.md]: `CD` was added in the second spot in the XML, so it wouldn't have a value of `7`

File: docs/Common/Enums/index.md [hunk 6]
```diff
@@ -925,6 +941,54 @@ Radio Buttons
...
+|FRONT_LEFT_DAYTIME_RUNNING_LIGHT|8||
+|FRONT_RIGHT_DAYTIME_RUNNING_LIGHT|9||
+|FRONT_LEFT_TURN_LIGHT|10||
+|FRONT_RIGHT_TURN_LIGHT|11||
+|REAR_LEFT_FOG_LIGHT|12||
+|REAR_RIGHT_FOG_LIGHT|13||
+|REAR_LEFT_TAIL_LIGHT|14||
+|REAR_RIGHT_TAIL_LIGHT|15||
+|REAR_LEFT_BRAKE_LIGHT|16||
+|REAR_RIGHT_BREAK_LIGHT|17||
```
Comment (jacobkeeler) [inline docs/Common/Enums/index.md]: `REAR_RIGHT_BREAK_LIGHT` needs to be changed as well

File: docs/Common/Enums/index.md (hunk 3 above)
Comment (KhrystynaDubovyk) [reply docs/Common/Enums/index.md]: @jacobkeeler, please clarify, should I remove it?

Comment (jacobkeeler) [reply docs/Common/Enums/index.md]: Yes

File: docs/Common/Structs/index.md [hunk 7]
```diff
@@ -468,6 +468,9 @@
 |moduleType|Common.ModuleType|true||The moduleType indicates which type of data should be changed and identifies which data object exists in this struct. For example, if the moduleType is CLIMATE then a "climateControlData" should exist|
 |radioControlData|Common.RadioControlData|false|||
 |climateControlData|Common.ClimateControlData|false||
+|audioControlData|Common.AudioControlData|false|||
```
Comment (jacobkeeler) [inline docs/Common/Structs/index.md]: Extra `|`, might be causing the parsing failure

File: docs/Common/Structs/index.md [hunk 8]
```diff
@@ -527,6 +549,9 @@
 |radioControlCapabilities|RadioControlCapabilities|false|array: true <br> minsize: 1 <br> maxsize: 100|If included, the platform supports RC radio controls. For this baseline version, maxsize=1. i.e. only one climate control module is supported|
 |ButtonCapabilities|false|array: true <br> minsize: 1 <br> maxsize: 100|If included, the platform supports RC button controls with the included button names|
 |seatControlCapabilities|Common.SeatControlCapabilities|false|minsize="1" <br> maxsize="100" <br> array="true"|If included, the platform supports seat controls.|
+|audioControlCapabilities|Common.AudioControlCapabilities|false| minsize="1" <br> maxsize="100" <br> array="true"| If included, the platform supports audio controls.|
+|hmiSettingsControlCapabilities|Common.HMISettingsControlCapabilities|false||If included, the platform supports hmi setting controls.|
+|lightControlCapabilities|Common.LightControlCapabilities|false|If included, the platform supports light controls.|
```
Comment (jacobkeeler) [inline docs/Common/Structs/index.md]: Missing `|` before description

File: docs/Common/Enums/index.md (hunk 5 above)
Comment (KhrystynaDubovyk) [reply docs/Common/Enums/index.md]: @jacobkeeler, fixed https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/103/commits/4212ef5c22a5dfe946b9c4d0d8a1d63040d171eb

File: docs/Common/Enums/index.md (hunk 6 above)
Comment (KhrystynaDubovyk) [reply docs/Common/Enums/index.md]: @jacobkeeler, fixed https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/103/commits/e7d66dee1aea44accd4d2caedf6dbf659a6bfb0b

File: docs/Common/Structs/index.md [hunk 9]
```diff
@@ -543,6 +568,96 @@
...
+
+### HMISettingsControlData
+
+|Name|Type|Mandatory|Additional|Description|
+|:---|:---|:--------|:---------|:----------|
+|||Corresponds to "HMI_SETTINGS" ModuleType||
+|displayMode|Common.DisplayMode|false||
+|temperatureUnit|Common.TemperatureUnit|false||
+|distanceUnit|Common.DistanceUnit|false||
+
```
[the remark this answers is not in the document]
Comment (KhrystynaDubovyk) [reply docs/Common/Structs/index.md]: @jacobkeeler, fixed https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/103/commits/bdd6aeb5b1fee04de457f17f9fc7d753d911c854

File: docs/Common/Structs/index.md (hunk 8 above)
Comment (KhrystynaDubovyk) [reply docs/Common/Structs/index.md]: @jacobkeeler, fixed https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/103/commits/26a6389877e57cfc3a1475760cae24b13831b177

File: docs/Common/Structs/index.md (hunk 7 above)
Comment (KhrystynaDubovyk) [reply docs/Common/Structs/index.md]: @jacobkeeler, fixed https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/103/commits/15bd7c4a57a236266a14c48d32cb63372fe12e36

File: docs/Common/Structs/index.md [hunk 10]
```diff
@@ -486,6 +486,7 @@
 |signalChangeThreshold|Integer|false|minvalue:0 maxvalue:100|If the signal strength falls below the set value for this parameter, the radio will tune to an alternative frequency|
 |radioEnable|Boolean|false||True if the radio is on, false is the radio is off|
 |state|Common.RadioState|false||||
+|sisData|SisData|false||Read-only Station Information Service (SIS) data provides basic information about the station such as call sign, as well as information not displayable to the consumer such as the station identification number|
```
Comment (jacobkeeler) [inline docs/Common/Structs/index.md]: `Common.SisData`

File: docs/Common/Enums/index.md (hunk 3 above)
Comment (jacobkeeler) [reply docs/Common/Enums/index.md]: Still present

Comment (KhrystynaDubovyk) [reply docs/Common/Enums/index.md]: @jacobkeeler fixed https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/103/commits/c1863e0adc0ed87d956975819c142c796914f281

File: docs/Common/Enums/index.md [hunk 11]
```diff
@@ -925,6 +941,54 @@ Radio Buttons
...
+|FRONT_RIGHT_TURN_LIGHT|11||
+|REAR_LEFT_FOG_LIGHT|12||
+|REAR_RIGHT_FOG_LIGHT|13||
+|REAR_LEFT_TAIL_LIGHT|14||
+|REAR_RIGHT_TAIL_LIGHT|15||
+|REAR_LEFT_BRAKE_LIGHT|16||
+|REAR_RIGHT_BRAKE_LIGHT|17||
+|REAR_LEFT_TURN_LIGHT|18||
+|REAR_RIGHT_TURN_LIGHT|19||
+|REAR_REGISTRATION_PLATE_LIGHT|20||
```
Comment (jacobkeeler) [inline docs/Common/Enums/index.md]: Missing common function groups for Exterior and Interior lights

File: docs/Common/Structs/index.md [hunk 12]
```diff
@@ -543,6 +569,95 @@
...
+|volume|Integer|false|minvalue="0" <br> maxvalue="100"|Reflects the volume of audio, from 0%-100%."
+|equalizerSettings|Common.EqualizerSettings|false|minvalue="1" <br> maxvalue="100"<br>array ="true"|Defines the list of supported channels (band) and their current/desired settings on HMI|    
+
+
+### LightCapabilities
+
+|Name|Type|Mandatory|Additional|Description|
+|:---|:---|:--------|:---------|:----------|
+|name|Common.LightName|true|||
+|Assuming light ON/OFF status is always available|||||
```
Comment (jacobkeeler) [inline docs/Common/Structs/index.md]: Remove this line

File: docs/Common/Structs/index.md [hunk 13]
```diff
@@ -543,6 +569,95 @@
...
+|distanceUnitAvailable|Boolean|false||Availability of the control of distance unit.|
+|temperatureUnitAvailable|Boolean|false||Availability of the control of temperature unit.|
+|displayModeUnitAvailable|Boolean|false||Availability of the control of HMI display mode.|
+
+
+### HMISettingsControlData
+
+|Name|Type|Mandatory|Additional|Description|
+|:---|:---|:--------|:---------|:----------|
+|||Corresponds to "HMI_SETTINGS" ModuleType||
```
Comment (jacobkeeler) [inline docs/Common/Structs/index.md]: Remove from table

File: docs/Common/Enums/index.md (hunk 11 above)
Comment (KhrystynaDubovyk) [reply docs/Common/Enums/index.md]: @jacobkeeler, please check https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/103/commits/6c7514a2ad895080784013535044f7bc9ce237c0

File: docs/Common/Structs/index.md (hunk 12 above)
Comment (KhrystynaDubovyk) [reply docs/Common/Structs/index.md]: @jacobkeeler, please see https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/103/commits/be221a1ec3d602d1322977b64d00bd95d79e8351

File: docs/Common/Structs/index.md (hunk 13 above)
Comment (KhrystynaDubovyk) [reply docs/Common/Structs/index.md]: @jacobkeeler, fixed https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/103/commits/d8acf14733d99a5cc00d9f35843323c339fa032c
